src/evict/evict_lru.c

```diff
@@ -1885,14 +1885,20 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
      */
     switch (btree->evict_start_type) {
     case WT_EVICT_WALK_NEXT:
+        if (btree->evict_ref == NULL && (++btree->linear_walk_restarts) & 1)
+            goto rand_next;
         break;
     case WT_EVICT_WALK_PREV:
         FLD_SET(walk_flags, WT_READ_PREV);
+        if (btree->evict_ref == NULL && (++btree->linear_walk_restarts) & 1)
```

Comment:
Should this be before the statement to set the flag?

Reply:
Done